pkg/ccl/changefeedccl/changefeedbase/sink_url.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package changefeedbase
+
+import (
+	"encoding/base64"
+	"net/url"
+	"strconv"
+
+	"github.com/cockroachdb/cockroach/pkg/util"
+	"github.com/cockroachdb/errors"
+)
+
+// SinkURL is a helper struct which for "consuming" URL query
+// parameters from the underlying URL.
+type SinkURL struct {
```

Comment:
Thanks for pulling the `SinkURL` move into its own commit. Could you also put the code that uses the new struct into this commit? I think it's important that every intermediate commit builds so that we don't break git bisecting. It will also make the third commit smaller and easier to review.